plugins/events-backend-module-bitbucket-cloud/src/router/BitbucketCloudEventRouter.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { EventParams, EventRouter } from '@backstage/plugin-events-node';
+
+/**
+ * Subscribes to the generic `bitbucketCloud` topic
+ * and publishes the events under the more concrete sub-topic
+ * depending on the `x-event-key` provided.
+ *
+ * Events of other topics will be ignored (should never happen).
+ *
+ * @public
+ */
+export class BitbucketCloudEventRouter extends EventRouter {
+  constructor() {
+    super('bitbucketCloud');
+  }
+
+  protected detectSubTopic(params: EventParams): string | undefined {
+    return params.metadata?.['x-event-key'] as string | undefined;
```

Comment:
re `x-event-key`:
Actually, I used this in combination with the `aws-sqs` module without the `http` module, using this from the message attributes instead of HTTP headers. Of course, there is still an underlying contract for the metadata required and when putting messages into the AWS SQS queue, the message attribute has to be set based on the HTTP headers received (managed externally).

In our case, I could not install the `http` module at all, could have no HTTP endpoint for events.

Reply:
Thanks for the background info! I am perfectly fine with keeping this as-is, since it's the less limiting setup - we can always simplify later as an optimization so to speak (potentially fewer installation steps or less config), it's functionally fine in its current state.